qdax/core/emitters/mees_emitter.py

```diff
@@ -0,0 +1,613 @@
+from dataclasses import dataclass
+from functools import partial
+from typing import Callable, Tuple
+
+import jax
+import jax.numpy as jnp
+import optax
+
+from qdax.core.containers.mapelites_repertoire import (
+    MapElitesRepertoire,
+    get_cells_indices,
+)
+from qdax.core.emitters.emitter import Emitter, EmitterState
+from qdax.types import Descriptor, ExtraScores, Fitness, Genotype, RNGKey
+
+
+@dataclass
+class MEESConfig:
+    """Configuration for the MAP-Elites-ES emitter."""
+
+    sample_number: int = 1000  # num of samples for gradient estimate
+    sample_sigma: float = 0.02
+    sample_mirror: bool = True  # if 1, use mirroring sampling
+    sample_rank_norm: bool = True  # if 1, use normalisation
+    num_optimizer_steps: int = 10  # frequency of archive-sampling
+    adam_optimizer: bool = True  # if 1 use ADAM, if 0 use SGD
+    learning_rate: float = 0.01
+    learning_rate_decay: float = 1.0  # only applied when no ADAM
+    l2_coefficient: float = 0.02  # coefficient for regularisation
+    novelty_nearest_neighbors: int = 10
+    use_explore: bool = True  # if 0, use only fitness gradient
+
+
+class MEESEmitterState(EmitterState):
+    """Emitter State for the MAP-Elites-ES emitter."""
```

Comment:
Same here, might be cleaner to put descriptions in the dosctrings :slightly_smiling_face:

Reply:
Also done in the last commits!